cli/src/commands/split.rs

```diff
@@ -230,3 +242,28 @@ The remainder will be in the second commit.
     tx.finish(ui, format!("split commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+/// Reads 'split.legacy-bookmark-behavior' from settings and prints a warning if
+/// the value is unset to alert the user to the behavior change.
+fn read_legacy_bookmark_behavior_setting(
+    settings: &jj_lib::settings::UserSettings,
+    ui: &Ui,
+) -> Result<bool, CommandError> {
+    match settings
+        .get_bool("split.legacy-bookmark-behavior")
+        .optional()?
+    {
+        Some(value) => Ok(value),
+        None => {
+            writeln!(
+                ui.warning_default(),
+                "`jj split` will stop moving bookmarks in the next release.\n\
+                     Run `jj config set --repo split.legacy-bookmark-behavior true` \
+                     to silence this message and keep the current behavior.\n\
+                     See https://github.com/jj-vcs/jj/issues/3419"
+            )?;
```

Comment:
nit: use a raw string here `r#...#;` as it makes the intended indentation clearer and less error-prone.

Reply:
Thanks good point. I decided to use three `writeln!` calls instead so that each line is prefixed by `Warning:`. I think this makes it a little easier to distinguish the warning message from the rest of the info printed by `jj split`. I briefly searched through the codebase and didn't see much precedence for multi-line warning messages, so this approach seems reasonable to me.